grype/db/internal/gormadapter/open.go

```diff
@@ -88,22 +134,64 @@ func Open(path string, options ...Option) (*gorm.DB, error) {
 		return nil, fmt.Errorf("unable to connect to DB: %w", err)
 	}
 
-	if cfg.write {
-		for _, sqlStmt := range writerStatements {
-			dbObj.Exec(sqlStmt)
-			if dbObj.Error != nil {
-				return nil, fmt.Errorf("unable to execute (%s): %w", sqlStmt, dbObj.Error)
-			}
+	if cfg.writable {
+		log.Trace("applying writable DB statements")
+		if err := applyStatements(dbObj, writerStatements); err != nil {
+			return nil, fmt.Errorf("unable to apply DB writer statements: %w", err)
+		}
+	}
+
+	if cfg.truncate && cfg.allowLargeMemoryFootprint {
```

Comment:
Is the memory footprint related to truncating?

Reply:
it is -- when we start with an empty DB we will be doing a lot of writing (and allowing for a larger memory footprint in those cases allows for writing to the DB much faster).